Lib/pydoc.py

```diff
@@ -1500,10 +1500,9 @@ def docroutine(self, object, name=None, mod=None, cl=None):
                 argspec = str(signature)
                 if realname == '<lambda>':
                     title = self.bold(name) + ' lambda '
-                    # XXX lambda's won't usually have func_annotations['return']
-                    # since the syntax doesn't support but it is possible.
-                    # So removing parentheses isn't truly safe.
-                    argspec = argspec[1:-1] # remove parentheses
+                    # Since lambda's cannot have a parenthesis in their signature,
+                    # it's safe to replace them.
+                    argspec = argspec.replace("(", "").replace(")", "")
```

Comment:
See issue for explanation
```suggestion
                if realname != '<lambda>':
                    argspec = str(signature)
                else:
                    title = self.bold(name) + ' lambda '
                    # Save a potential return annotation
                    ret_ann = signature.return_annotation
                    # Clear the return annotation from the signature
                    signature._return_annotation = signature.empty
                    # Since lambda's cannot have parentheses in their signature,
                    # it's safe to remove them.
                    argspec = str(signature)[1:-1] # remove parentheses
                    if ret_ann is not signature.empty:
                        ret_ann = inspect.formatannotation(ret_ann)
                        argspec += f' -> {ret_ann}'
````

Reply:
Great!
One question: are you sure that we should delete the parenthesis from the annotations?
With this solution, parenthesis will be deleted from annotations:
```python
func = lambda a, b, c: 1
func.__annotations__ = {"return": (int, int)}
help(func)
>
Help on function <lambda> in module __main__:

<lambda> lambda a, b, c -> <class 'int'>, <class 'int'>
```
It's probably ok, but I don't sure.

Reply:
Great! 
I'll add a test case for situation which you described in issue.

Reply:
Oh, I can't apply your suggestion..
![image](https://github.com/python/cpython/assets/80244920/6a1f1693-6932-46b9-a66d-433567a97441)

Reply:
The GitHub interface has its quirks -- I pushed a revised suggestion to your branch, hope it is useful!